build/charts/flow-aggregator/values.yaml

```diff
@@ -45,8 +45,8 @@ flowCollector:
 clickHouse:
   # -- Determine whether to enable exporting flow records to ClickHouse.
   enable: false
-  # -- DatabaseURL is the url to the database. TCP protocol is required.
-  databaseURL: "tcp://clickhouse-clickhouse.flow-visibility.svc:9000"
+  # -- DatabaseURL is the url to the database. Currently only TCP is supported.
+  databaseURL: "clickhouse-clickhouse.flow-visibility.svc:9000"
```

Comment:
Do we intend to support https as well later?
BTW, if one explicitly adds the `tcp://` scheme to the URL, does it still work?

Reply:
yes, I am planning to do https and tls support in another PR.

It will not work if someone try to add protocol to it. Do you think we should add a comment here to clarify it?

Reply:
I don't think a comment is necessary, but the PR description and comment should make it clear that we have another non-backward compatible change.
Also before we merge this, I want to understand what the configuration changes will be to enabled https support. Maybe we could make the changes as part of this PR, even if we don't implement that support yet. And just to clarify: do we ever intend to support the native CH interface as well?

Reply:
I'm thinking of something like this to be added to FA config map:
```yaml
# -- Protocol used by connection between FlowAggregator and ClickHouse. It must
# be one of "tcp", "tls", "http", "https".
protocol: "tcp"
```

no, native CH interface is introduced from clickhouse-go/v2. In the our current clickhouse client implementation, we are just using `database/sql` interface and calling `sql.open()`. Not sure if I answered your question. From user perspective, I don't think the interface choosing will make a difference?

Reply:
> no, native CH interface is introduced from clickhouse-go/v2

what's your point here? We are updating to v2 after all

> From user perspective, I don't think the interface choosing will make a difference?

The native interface has better performance according to https://github.com/ClickHouse/clickhouse-go#benchmark. That being said I am not sure it matters. If we ever want to add support for this interface, it would be nice to have a way to do that without breaking backwards compatibility for the user. But I guess what we could do is change the implementation for "tcp" / "tls" from std to native interface. An alternative would be introducing a boolean flag if we ever need it.

Reply:
I forgot the CA certificate. TBH I haven't done enough investigation yet. I suppose it would like:

```yaml
# Protocol used by connection between FlowAggregator and ClickHouse. It must
# be one of "tcp", "tls", "http", "https".
protocol: "tcp"
# Indicates whether to use default root CAs, if protocol is chosen to be "tls" or "https".
# If false, a Secret named "antrea-clickhouse-ca" must be provided with the following key:
#   tls.crt: <CA certificate>
selfSignedCA: false
```

Reply:
> it would be nice to have a way to do that without breaking backwards compatibility for the user

I see your point. Let me have a try.

Reply:
Just to clarify, I am not saying we should implement this now. But in the future we will need to avoid making breaking changes to the config.
This discussion is a bit orthogonal to this PR. It seems that most of the config changes will happen when we add HTTP & TLS support, so this is when we will need to be careful. It seems that the protocol will be specified through a different config field, and not through the database URL, which is fine by me.

Reply:
Sorry Antonin, I think more and I started to doubt whether I got your point right. Let me clarify: 

What we want to achieve? - backward compatibility + better performance on tcp/tls, right?

> An alternative would be introducing a boolean flag if we ever need it.

I didn't get what is this boolean flag for?

- backward compatibility - We can strip the `tcp://` scheme if we find it in the `databaseURL`. It's a bit hacky, but maybe it is the only way to support backward compatibility, unless we use std interface with DSN connection, like what we do in the current FA.

- Changing the implementation for "tcp/tls" from std to native interface - I think it can take some time. It is because native interface returns a `clickhouse.Conn`, and std interface returns `sql.db`. For db writes, we need to replace the bulk write: begin->prepare->(in loop exec)->commit to `conn.Exec`, maybe also make use of the batch API. Most of the functions, along with UTs need to be rewritten. I currently have other things to work on. It depends on how we evaluate the priorities between changing to native interface and adding tls&http support.

Reply:
Another question is - 

> Also before we merge this, I want to understand what the configuration changes will be to enabled https support. Maybe we could make the changes as part of this PR, even if we don't implement that support yet

are you asking about https or http? If http, do you think adding the following lines would be ok?
```yaml
# Protocol used by connection between FlowAggregator and ClickHouse. It must
# be one of "tcp", "http".
protocol: "tcp"
```
If https, I think we also need to add the CA cert option? Do we want to include the FA configurations into this PR?

Reply:
Please ignore my last two comments - I didn't see your clarification when I wrote those.

> It seems that the protocol will be specified through a different config field, and not through the database URL, which is fine by me.

It actually makes much more sense to me if we retrieve the protocol from database URL. I will push a new commit with this change.

Reply:
> It actually makes much more sense to me if we retrieve the protocol from database URL. I will push a new commit with this change.

Is it because it is more consistent with the FlowExporter configuration?
I don't have a strong preference either way. Let's see what others have to say when they review.

Reply:
yes, that's one of the reason. It also avoid breaking changes to the config.